app/test/app.test.js

```diff
@@ -0,0 +1,57 @@
+const request = require('sync-request');
+const postModel = require('../src/model/post');
+
+test('check if title is string type', async () => {
+    let result = request('POST', 'http://localhost:8080/posts', {
+        json: {
+            title: '<string>',
+            slug: '<string>',
+            content: '<string>',
+        }
+    });
+    expect(result.statusCode).toBe(201);
+    let payload = JSON.parse(result.body);
+    expect(typeof payload.id).toBe('number');
+    await postModel.deletePost(payload);
+});
+
+test('check if title is string type', () => {
+    let result = request('POST', 'http://localhost:8080/posts', {
+        json: {
+            title: 2,
+            slug: '<string>',
+            content: 3,
+        }
+    });
+    expect(result.statusCode).toBe(400);
+    let payload = JSON.parse(result.body);
+    expect(payload.Errors).toEqual(["title is not a string type", "content is not a string type"]);
+});
+
+
+test('check if the slug exist --> does not exist', async () => {
+    let result = request('POST', 'http://localhost:8080/posts', {
+        json: {
+            title: '1',
+            slug: '2',
+            content: '3'
+        }
+    });
+    let payload = JSON.parse(result.body);
+
+
+    let result1 = request('POST', 'http://localhost:8080/posts', {
+        json: {
+            title: '1',
+            slug: '2',
+            content: '3'
+        }
+    });
+    let payload1 = JSON.parse(result.body);
+    expect(result1.statusCode).toBe(400);
+
+    if (await postModel.isSlugExist(result1)) {
+        await postModel.deletePost(payload);
+    }
+    await postModel.deletePost(payload1);
```

Comment:
`payload1` nie zawiera encji dlatego tutaj spowodujesz błąd

Reply:
w pełni ufam temu, co mówisz mam tylko pytanie, jak usuwam tego payloada1, to post zapisuje mi się w bazie. jak mogę to inaczej rozegrać aby wyglądało poprawnie?

Reply:
Nie zapisuje się, bo aplikacja rzuca błąd walidacji

Reply:
poprawione